invectio/lib.py

```diff
@@ -148,6 +155,142 @@ def get_module_report(self) -> dict:
+                class_def.name,
+            )
+
+        self.symbols.add(class_def.name)
+
+    def visit_Assign(self, assign: ast.Assign) -> None:
+        """Visit a global."""
+
+        def _maybe_add_maybe_log(n: str):
+            if not self.include_private and n.startswith("_"):
+                return
+
+            if n in self.symbols:
+                _LOGGER.warning(
+                    "Target in assignment overrides already defined symbol in file %r: %r",
+                    self.file_name,
+                    n,
+                )
+
+            self.symbols.add(n)
+
+        def _maybe_add(node: ast.AST):
+            if isinstance(node, ast.Name):
+                _maybe_add_maybe_log(node.id)
+            elif isinstance(node, ast.Starred):
+                _maybe_add(node.value)
+            elif isinstance(node, ast.Tuple):
+                for i in node.elts:
+                    _maybe_add(i)
+            elif isinstance(node, (ast.Subscript, ast.Attribute)):
+                # Should be declared beforehand.
+                pass
+            else:
+                _LOGGER.error(
+                    "Unhandled type for target in assignment: %r",
+                    node.__class__.__name__,
+                )
+
+        for target in assign.targets:
+            if isinstance(target, ast.Name):
+                _maybe_add_maybe_log(target.id)
+            else:
+                try:
+                    _maybe_add(target)
+                except RecursionError:
+                    _LOGGER.exception(
+                        f"Failed to parse assign statement in {self.file_name}"
+                    )
+
+    def visit_AnnAssign(self, ann_assign: ast.AugAssign) -> None:
+        """Visit aug assignments."""
+        if not isinstance(ann_assign.target, ast.Name):
+            # Skip subscription and attributes.
+            return
+
+        if not self.include_private and ann_assign.target.id.startswith("_"):
+            return
+
+        if ann_assign.target.id in self.symbols:
+            _LOGGER.warning(
+                "Target in assignment overrides already defined symbol in file %r: %r",
+                self.file_name,
+                ann_assign.target.id,
+            )
+
+        self.symbols.add(ann_assign.target.id)
+
+    def visit(self, module: ast.Module) -> None:
+        """Gather symbols provided by the given ast.
+
+        Note we are traversing top level modules exported. We do not visit nested AST trees.
+        """
+        for item in module.body:
+            handler_name = f"visit_{item.__class__.__name__}"
+            handler = getattr(self, handler_name, None)
+            if handler:
+                handler(item)
+
+    def get_module_report(self) -> Set[str]:
+        """Get report once the traversal is done."""
+        module_name = (
+            self.file_name[: -len(".py")]
+            if self.file_name.endswith(".py")
+            else self.file_name
```

Comment:
When don't python files end in `.py`? or am I missing something?

Reply:
yeah, the check is probably redundant